src/peripherals.rs

```diff
@@ -102,6 +104,8 @@ pub struct Peripherals {
         any(esp_idf_version_major = "4", esp_idf_version = "5.0")
     ))]
     pub twdt: watchdog::TWDT,
+    #[cfg(esp_idf_soc_usb_serial_jtag_supported)]
```

Comment:
* As per my summary, I think we need to additionally conditionalize the exposure of this peripheral depending on whether it is "grabbed" by the VFS layer or not. If it IS grabbed by the VFS layer (and the VFS component is enabled/selected in the first place), this peripheral should not be available in the `Peripherals` struct. 
* Ditto for the UART0/1 peripherals